### DISCUSSION TITLE
Not a registered object error

### DISCUSSION THREAD

**Discussion by mohbarka** [idaholab/moose#16921] [Q&A Modules: General] [answered]
Good mooring Moose community, 

I am trying to create  a new  kernel for coupled  Moisture chloride diffusion in concrete and. when I run the  example  I get this error 
** ERROR ***
A 'ConcreteMoistureTimeIntegration' is not a registered object.
If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable

Can anyone have any idea what could be wrong?  below show what I have  in both C ,and f files 

Thank you!
 

C file 

#include "ConcreteMoistureTimeIntegration.h"
#include "Material.h"
registerMooseObject("MooseTestApp", ConcreteMoistureTimeIntegration);
template <>
InputParameters
validParams<ConcreteMoistureTimeIntegration>()
{
  InputParameters params = validParams<TimeDerivative>();
  return params;
}

h file 

#pragma once
#include "TimeDerivative.h"
// Forward Declaration
class ConcreteMoistureTimeIntegration;
template <>
InputParameters validParams<ConcreteMoistureTimeIntegration>();

**Comment by aeslaughter** [comment, marked as answer]
You are registering to "MooseTestApp", but you should be registering it to the application you created.

https://mooseframework.inl.gov/getting_started/new_users.html

**Reply by aaaaaaqing** [reply]
cant understand u say，what should we do
